pkgs/development/python-modules/pdm-backend/default.nix

```diff
@@ -40,6 +42,10 @@ buildPythonPackage rec {
     setuptools
   ];
 
+  preCheck = ''
+    unset PDM_BUILD_SCM_VERSION
+  '';
```

Comment:
What's the purpose of this?

Reply:
I need to set the variable during build, but unset it during tests, so it won't affect the version related tests.

Reply:
If setting it during build is supported by upstream, they should also sanitize it when running tests?

Reply:
TBH it's not their fault we are carrying this environment variable throughout the whole build instead of passing it only to the build process.